core/test/base/mtx_io.cpp

```diff
@@ -870,7 +870,7 @@ class DummyLinOp
       public gko::EnableCreateMethod<DummyLinOp<ValueType, IndexType>>,
       public gko::ReadableFromMatrixData<ValueType, IndexType>,
       public gko::WritableToMatrixData<ValueType, IndexType> {
-    friend class gko::EnablePolymorphicObject<DummyLinOp, gko::LinOp>;
+    friend class gko::polymorphic_object_traits<DummyLinOp>;
```

Comment:
Are you sure this works as expected ? This class no longer has the `copy_from`, `clone` etc impls that `EnableAbstractPolymorphicObject` provided through the `EnablePolymorphicObject` interface.

Reply:
The classes still derive from `EnableAbstractPolymorphicObject` somewhere in the hierarchy. This only changes which class has access to the protected methods, mainly the constructor.

Reply:
As far as I can see, only `LinOp`, `ReorderingBase` and `Criterion` and the `AbstractFactory` derive from `EnableAbstractPolymorphicObject`. So, any classes that derive from these should be okay. But the rest will not be ?